mesonbuild/interpreter/interpreter.py

```diff
@@ -3206,10 +3207,24 @@ def build_library(self, node, args, kwargs):
         else:
             raise InterpreterException(f'Unknown default_library value: {default_library}.')
 
+    @staticmethod
+    def __extract_language_args(kwargs: T.Dict[str, T.List[mesonlib.FileOrString]]) -> None:
+        """Convert split language args into a combined dictionary.
+
+        The Meson DSL takes arguments in the form `<lang>_args : args`, but in the
+        build layer we store these in a single dictionary as `{<lang>: args}`.
+        This function extracts the arguments from the DSL format and prepares
+        them for the IR.
+        """
+        args: T.DefaultDict[str, T.List[mesonlib.FileOrString]] = collections.defaultdict(list)
+        for l in compilers.all_languages:
+            lang = kwargs[f'{l}_args']
+            args[l].extend(lang)
```

Comment:
Isn't this just:
```python
args[l] = kwargs[f'{l}_args']
```

Reply:
no, because extend gives you a copy, where that makes a reference. Which is really important for `both_library()`

Reply:
A copy of what? `extend()` does not make a copy of the list it is extending, it it would, you would need to assign the return value of `extend()` to something.

Reply:
Oh I see... we are extending it in this PR, indeed.

I think in that case,

```python
args[l] = kwargs[f'{l}_args'].copy()
```

looks nicer -- and timeit says it's faster, too...

Reply:
@dnicolodi, the point is that if we extend the default list here then it creates a copy of the argument to extend in the process of adding its contents in-place to the list being extended.

Reply:
Thanks @eli-schwartz I think I understood the point now. Another upside of your approach is that a defaultdict is not needed if you use the copy and assignment and not the extend.

Reply:
So I can either leave `.extend()` here with a comment, or move to `args[l] = lang.copy()`, but we have pretty deep assumptions that the language args we are given are owned by the BuildTarget class, so we need to do one or the other.